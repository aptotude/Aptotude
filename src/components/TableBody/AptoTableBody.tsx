import React from 'react';
import { StandardTypes } from '../../utils/standardTypes';
import classNames from 'classnames';

const COMPONENT_PREFIX = 'AptoTableBody';

export class AptoTableBody extends React.PureComponent<StandardTypes> {
  public render() {
    const {
      children,
      ...rest
    } = this.props;

    const classes = classNames(
      COMPONENT_PREFIX
    );

    return (
      <tbody className={classes} {...rest}>
        {children}
      </tbody>
    );
  }
}
