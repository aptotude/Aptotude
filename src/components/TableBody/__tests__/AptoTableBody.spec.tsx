import * as React from 'react';
import { render, cleanup } from 'react-testing-library';
import { AptoTableBody } from '../AptoTableBody';

afterEach(() => {
  cleanup();
});

describe('AptoTableBody', () => {
  it('Renders Component', () => {
    const { container } = render(
      <table>
        <AptoTableBody>
          <tr>
            <td>hi</td>
          </tr>
        </AptoTableBody>
      </table>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders extra classes', () => {
    const { container } = render(
      <table>
        <AptoTableBody className="foo">
          <tr>
            <td>hi</td>
          </tr>
        </AptoTableBody>
      </table>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
