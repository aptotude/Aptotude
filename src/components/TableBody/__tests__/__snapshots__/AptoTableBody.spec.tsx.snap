// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AptoTableBody Renders Component 1`] = `
<table>
  <tbody
    class="AptoTableBody"
  >
    <tr>
      <td>
        hi
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTableBody Renders extra classes 1`] = `
<table>
  <tbody
    class="foo"
  >
    <tr>
      <td>
        hi
      </td>
    </tr>
  </tbody>
</table>
`;
